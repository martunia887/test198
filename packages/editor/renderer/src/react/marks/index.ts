import { ComponentType } from 'react';
import { Mark } from 'prosemirror-model';

import Alignment from './alignment';
import Annotation from './annotation';
import Breakout from './breakout';
import Code from './code';
import ConfluenceInlineComment from './confluence-inline-comment';
import Em from './em';
import Indentation from './indentation';
import Link from './link';
import Strike from './strike';
import Strong from './strong';
import Subsup from './subsup';
import TextColor from './textColor';
import Underline from './underline';

export const markToReact: { [key: string]: ComponentType<any> } = {
  code: Code,
  em: Em,
  link: Link,
  strike: Strike,
  strong: Strong,
  subsup: Subsup,
  textColor: TextColor,
  underline: Underline,
  annotation: Annotation,

  // Stage0
  confluenceInlineComment: ConfluenceInlineComment,
  breakout: Breakout,
  alignment: Alignment,
  indentation: Indentation,
};

export const toReact = (mark: Mark): ComponentType<any> => {
  return markToReact[mark.type.name];
};

export {
  Code,
  Em,
  Link,
  Strike,
  Strong,
  Subsup,
  TextColor,
  Underline,
  Breakout,
  Annotation,
};
