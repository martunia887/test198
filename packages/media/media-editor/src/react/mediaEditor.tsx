import * as React from 'react';
import Spinner from '@atlaskit/spinner';
import {
  MediaEditorContainer,
  OutputArea,
  DrawingCanvas,
  HiddenTextArea,
  HiddenTextHelperDiv,
  SupplementaryCanvas,
  SpinnerWrapper,
  SketchBackground,
} from './styled';
import { Engine } from '../engine/engine';
import {
  ColorWithAlpha,
  Dimensions,
  ExportedImage,
  ShapeParameters,
  TextDirection,
  Tool,
} from '../common';
import { colorSame, colorWithAlphaSame, dimensionsSame } from '../util';

import {
  DefaultDrawingArea,
  OutputSize,
} from '../engine/components/drawingArea';
import {
  DefaultImageProvider,
  urlImageLoader,
} from '../engine/components/imageProvider';
import { DefaultMouseInput } from '../engine/components/mouseInput';
import { DefaultToolbar } from '../engine/components/toolbar';
import { DefaultKeyboardInput } from '../engine/components/keyboardInput';
import { DefaultImageReceiver } from '../engine/components/imageReceiver';
import { DefaultShapeDeleter } from '../engine/components/shapeDeleter';

export const DEFAULT_WIDTH = 845;
export const DEFAULT_HEIGHT = 530;
export const TOOLBAR_HEIGHT = 64;
export const DEFAULT_RATIO = DEFAULT_HEIGHT / DEFAULT_WIDTH;

export type ImageGetter = (format?: string) => ExportedImage;

export interface LoadParameters {
  imageGetter: ImageGetter;
}

export type LoadHandler = (
  imageUrl: string,
  loadParameters: LoadParameters,
) => void;
export type ErrorHandler = (imageUrl: string, error: Error) => void;
export type ShapeParametersChangedHandler = (
  parameters: ShapeParameters,
) => void;

export interface MediaEditorProps {
  imageUrl: string;
  dimensions: Dimensions;
  screenScaleFactor?: number;
  backgroundColor: ColorWithAlpha;
  shapeParameters: ShapeParameters;
  tool: Tool;

  onLoad: LoadHandler;
  onError: ErrorHandler;
  onShapeParametersChanged: ShapeParametersChangedHandler;
  onAnyEdit?: () => void;
  isEmptySketch?: boolean;
}

export interface MediaEditorState {
  isImageLoaded: boolean;
}

const defaultTextDirection = 'ltr';

export class MediaEditor extends React.Component<
  MediaEditorProps,
  MediaEditorState
> {
  private isUnmounted: boolean;

  // DOM elements that we need to create the engine
  private outputArea!: HTMLDivElement;
  private canvas!: HTMLCanvasElement;
  private supplementaryCanvas!: HTMLCanvasElement;
  private hiddenTextArea!: HTMLTextAreaElement;
  private hiddenTextHelperDiv!: HTMLDivElement;

  // Engine and its components
  private drawingArea?: DefaultDrawingArea;
  private toolbar?: DefaultToolbar;
  private engine?: Engine;

  constructor(props: MediaEditorProps) {
    super(props);
    this.isUnmounted = false;
    this.state = {
      isImageLoaded: false,
    };
  }

  componentDidMount() {
    this.loadEngine();
  }

  componentDidUpdate(prevProps: MediaEditorProps) {
    if (!this.engine) {
      return;
    }

    const currProps = this.props;

    if (currProps.imageUrl !== prevProps.imageUrl) {
      this.unloadEngine();
      this.loadEngine();
    }

    if (
      this.drawingArea &&
      (!dimensionsSame(currProps.dimensions, prevProps.dimensions) ||
        currProps.screenScaleFactor !== prevProps.screenScaleFactor)
    ) {
      this.drawingArea.setSize(MediaEditor.toOutputSize(currProps));
    }

    if (
      !colorWithAlphaSame(currProps.backgroundColor, prevProps.backgroundColor)
    ) {
      // TODO inform the core about the new background color
      // https://jira.atlassian.com/browse/FIL-3996
    }

    const {
      color: currColor,
      lineWidth: currLineWidth,
      addShadow: currAddShadow,
    } = currProps.shapeParameters;
    const {
      color: prevColor,
      lineWidth: prevLineWidth,
      addShadow: prevAddShadow,
    } = prevProps.shapeParameters;
    if (this.toolbar) {
      if (!colorSame(currColor, prevColor)) {
        this.toolbar.setColor(currColor);
      }
      if (currLineWidth !== prevLineWidth) {
        this.toolbar.setLineWidth(currLineWidth);
      }
      if (currAddShadow !== prevAddShadow) {
        this.toolbar.setAddShadow(currAddShadow);
      }
      if (currProps.tool !== prevProps.tool) {
        this.toolbar.setTool(currProps.tool);
      }
    }
  }

  componentWillUnmount() {
    this.isUnmounted = true;
    this.unloadEngine();
  }

  private handleOutputAreaInnerRef = (outputArea: HTMLDivElement) => {
    this.outputArea = outputArea;
  };
  private handleSupplementaryCanvasInnerRef = (canvas: HTMLCanvasElement) => {
    this.supplementaryCanvas = canvas;
  };
  private handleHiddenTextAreaInnerRef = (textArea: HTMLTextAreaElement) => {
    this.hiddenTextArea = textArea;
  };
  private handleHiddenTextHelperDivInnerRef = (div: HTMLDivElement) => {
    this.hiddenTextHelperDiv = div;
  };
  private handleDrawingCanvasInnerRef = (canvas: HTMLCanvasElement) => {
    this.canvas = canvas;
  };

  private renderSpinner = () => (
    <SpinnerWrapper>
      <Spinner size="large" invertColor={true} />
    </SpinnerWrapper>
  );

  private get sketchDimensions() {
    const { dimensions } = this.props;
    const sketchDimensions = {
      width: DEFAULT_WIDTH,
      height: DEFAULT_HEIGHT - TOOLBAR_HEIGHT,
    };
    if (dimensions.height < sketchDimensions.height) {
      const extraPadding = 10;
      sketchDimensions.width =
        (dimensions.height - extraPadding) / DEFAULT_RATIO;
      sketchDimensions.height = dimensions.height - extraPadding;
    }
    if (dimensions.width < sketchDimensions.width) {
      const extraPadding = 20;
      sketchDimensions.height =
        (dimensions.width - extraPadding) * DEFAULT_RATIO;
      sketchDimensions.width = dimensions.width - extraPadding;
    }
    return sketchDimensions;
  }

  render() {
    const { isImageLoaded } = this.state;
    const { dimensions, isEmptySketch } = this.props;

    return (
      <MediaEditorContainer style={dimensions}>
        {!isImageLoaded ? this.renderSpinner() : null}
        <OutputArea innerRef={this.handleOutputAreaInnerRef} style={dimensions}>
          <SupplementaryCanvas
            innerRef={this.handleSupplementaryCanvasInnerRef}
          />

          <HiddenTextArea
            autoComplete={'off'}
            innerRef={this.handleHiddenTextAreaInnerRef}
          />

          <HiddenTextHelperDiv
            innerRef={this.handleHiddenTextHelperDivInnerRef}
          />
          {isEmptySketch ? (
            <SketchBackground style={this.sketchDimensions} />
          ) : null}
          <DrawingCanvas
            onClick={this.onCanvasClick}
            innerRef={this.handleDrawingCanvasInnerRef}
            style={dimensions}
          />
        </OutputArea>
      </MediaEditorContainer>
    );
  }

  private onCanvasClick = () => {
    const { onAnyEdit } = this.props;
    if (onAnyEdit) {
      onAnyEdit();
    }
  };

  private loadEngine(): void {
    const { imageUrl, isEmptySketch } = this.props;

    DefaultImageProvider.create(
      () => urlImageLoader(imageUrl),
      this.supplementaryCanvas,
    )
      .then(imageProvider => {
        // We must not create the engine if the component was unmounted or if the image was changed
        if (this.isUnmounted || imageUrl !== this.props.imageUrl) {
          return;
        }
        this.setState({ isImageLoaded: true });

        if (isEmptySketch) {
          const sketchDimensions = this.sketchDimensions;
          imageProvider.backImage.width = sketchDimensions.width;
          imageProvider.backImage.height = sketchDimensions.height;
        }

        // Creating components for the engine
        const outputSize = MediaEditor.toOutputSize(this.props);
        const { backgroundColor } = this.props;
        this.drawingArea = new DefaultDrawingArea(
          this.canvas,
          outputSize,
          backgroundColor,
        );

        const mouseInput = new DefaultMouseInput(this.outputArea);
        this.toolbar = new DefaultToolbar(params =>
          this.props.onShapeParametersChanged(params),
        );
        const keyboardInput = new DefaultKeyboardInput(
          this.hiddenTextArea,
          this.supplementaryCanvas,
          this.hiddenTextHelperDiv,
        );
        const imageReceiver = new DefaultImageReceiver(
          this.supplementaryCanvas,
        );
        const shapeDeleter = new DefaultShapeDeleter(this.hiddenTextArea);

        // Creating the engine
        const { shapeParameters, tool: initialTool } = this.props;
        const textDirection =
          (window.getComputedStyle(this.outputArea)
            .direction as TextDirection) || defaultTextDirection;

        const config = {
          // tslint:disable-next-line:no-console
          onCoreError: (message: string) => {
            // tslint:disable-next-line
            console.error(message);
          },
          shapeParameters,
          initialTool,
          textDirection,
          drawingArea: this.drawingArea,
          imageProvider,
          mouseInput,
          toolbar: this.toolbar,
          keyboardInput,
          imageReceiver,
          shapeDeleter,
        };

        this.engine = new Engine(config);
        const loadParameters: LoadParameters = {
          imageGetter: (format?: string) => this.engine!.getBase64Image(format),
        };

        this.props.onLoad(imageUrl, loadParameters);
      })
      .catch(error => this.props.onError(imageUrl, error));
  }

  private unloadEngine(): void {
    if (this.engine) {
      this.engine.unload();
      delete this.engine;
      this.setState({ isImageLoaded: false });
    }
  }

  private static toOutputSize(props: MediaEditorProps): OutputSize {
    const { dimensions } = props;
    const screenScaleFactor =
      props.screenScaleFactor || MediaEditor.screenScaleFactor;

    return {
      width: dimensions.width * screenScaleFactor,
      height: dimensions.height * screenScaleFactor,
      screenScaleFactor,
    };
  }

  private static get screenScaleFactor(): number {
    return window.devicePixelRatio || 1;
  }
}

export default MediaEditor;
