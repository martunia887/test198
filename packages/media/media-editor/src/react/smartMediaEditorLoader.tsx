import * as React from 'react';

import { ModalSpinner } from '@atlaskit/media-ui';
import { colors } from '@atlaskit/theme';
import {
  SmartMediaEditorOwnProps,
  SmartMediaEditorProps,
} from './smartMediaEditor';
import {
  withMediaClient,
  WithOptionalMediaClientProps,
} from '@atlaskit/media-client';

interface State {
  SmartMediaEditor?: React.ComponentType<SmartMediaEditorProps>;
}

export type Props = SmartMediaEditorOwnProps & WithOptionalMediaClientProps;

export class SmartMediaEditorLoader extends React.PureComponent<
  Props,
  State & { isErrored: boolean }
> {
  static displayName = 'AsyncSmartMediaEditor';
  static SmartMediaEditor?: React.ComponentType<SmartMediaEditorProps>;

  state = {
    // Set state value to equal to current static value of this class.
    SmartMediaEditor: SmartMediaEditorLoader.SmartMediaEditor,
    isErrored: false,
  };

  async componentWillMount() {
    if (!this.state.SmartMediaEditor) {
      try {
        const module = await import(/* webpackChunkName:"@atlaskit-internal_smart-media-editor" */
        './smartMediaEditor');
        SmartMediaEditorLoader.SmartMediaEditor =
          module.SmartMediaEditorWithIntl;
        this.setState({ SmartMediaEditor: module.SmartMediaEditorWithIntl });
      } catch (e) {
        // tslint:disable-next-line:no-console
        console.error(e);
        this.setState({ isErrored: true });
      }
    }
  }

  render() {
    const { isErrored } = this.state;
    const { mediaClient } = this.props;

    if (isErrored) {
      return null;
    }
    if (!this.state.SmartMediaEditor || !mediaClient) {
      return (
        <ModalSpinner blankedColor={colors.N700A} invertSpinnerColor={true} />
      );
    }

    return (
      <this.state.SmartMediaEditor {...this.props} mediaClient={mediaClient} />
    );
  }
}

export default withMediaClient(SmartMediaEditorLoader);
