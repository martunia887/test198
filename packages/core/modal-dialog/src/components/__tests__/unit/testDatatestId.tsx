import React from 'react';
import { mount } from 'enzyme';
import { render, cleanup } from '@testing-library/react';

import cases from 'jest-in-case';

import ModalDialog, { ModalTransition } from '../../..';

// dialogs require an onClose function
const noop = () => {};

const MyContent = () => <div>Hello</div>;
const wrapperWithTestId = (
  <ModalTransition>
    <ModalDialog onClose={noop} testId="iamTheDataTestId">
      <MyContent />
    </ModalDialog>
  </ModalTransition>
);

describe('Using enzyme', () => {
  test('It should not generate data-testid', () => {
    const wrapper = mount(
      <ModalTransition>
        <ModalDialog onClose={noop}>
          <MyContent />
        </ModalDialog>
      </ModalTransition>,
    );
    expect(wrapper).toBeDefined();
    expect(wrapper.text()).toBe('Hello');
    expect(wrapper.prop('data-testid')).toBeUndefined();
    wrapper.unmount();
  });
  test('Modal-dialog snapshot should be same with data-testid ', () => {
    const wrapper = mount(wrapperWithTestId);
    expect(
      wrapper.find('div[data-testid="iamTheDataTestId"]'),
    ).toMatchSnapshot();
    wrapper.unmount();
  });

  describe('Modal-dialog with different data-testid', () => {
    cases(
      'should be generated',
      ({ key }: { key: string }) => {
        const wrapper = mount(wrapperWithTestId);
        expect(wrapper.find(`[data-testid='${key}']`)).toBeTruthy();
        wrapper.unmount();
      },
      [
        { key: 'josefGiTan' },
        { key: 'ZZZZŹŽ;;;;' },
        { key: '@3$&&&&Helooo' },
        { key: '126^^^' },
        { key: 123 },
      ],
    );
  });
});

describe('Using react-test-library', () => {
  describe('Modal should be found by data-testid', () => {
    beforeEach(() => cleanup());
    test('Using getByTestId()', async () => {
      const { getByTestId } = render(wrapperWithTestId);
      expect(getByTestId('iamTheDataTestId')).toBeTruthy();
    });

    test('Using container snapshot', () => {
      const { container } = render(wrapperWithTestId);
      expect(container).toMatchSnapshot();
    });
  });
});
