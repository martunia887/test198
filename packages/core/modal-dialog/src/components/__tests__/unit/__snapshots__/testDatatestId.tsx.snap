// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Using enzyme Modal-dialog snapshot should be same with data-testid  1`] = `
.emotion-3 {
  background-color: #FFFFFF;
  border-radius: 3px;
  box-shadow: 0 0 0 1px rgba(9,30,66,0.08),0 2px 1px rgba(9,30,66,0.08), 0 0 20px -6px rgba(9,30,66,0.31);
  color: #091E42;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: auto;
  max-height: 100%;
  outline: 0;
  pointer-events: auto;
}

@media only screen and (-ms-high-contrast:active),(-ms-high-contrast:none) {
  .emotion-3 {
    max-height: calc(100% - 1px);
  }
}

@media (min-width:320px) and (max-width:480px) {
  .emotion-3 {
    height: 100%;
    max-height: 100%;
    border-radius: 0;
  }
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  max-height: 100%;
}

@media only screen and (-ms-high-contrast:active),(-ms-high-contrast:none) {
  .emotion-2 {
    max-height: calc(100% - 1px);
  }
}

.emotion-1 {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  overflow-y: auto;
  overflow-x: hidden;
  padding: 2px 24px;
}

@media (min-width:320px) and (max-width:480px) {
  .emotion-1 {
    overflow-y: auto;
    height: 100%;
  }
}

.emotion-0 {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  padding: 0 24px;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  overflow-y: auto;
  overflow-x: hidden;
  padding: 2px 24px;
}

@media (min-width:320px) and (max-width:480px) {
  .emotion-0 {
    overflow-y: auto;
    height: 100%;
  }
}

@media (min-width:320px) and (max-width:480px) {
  .emotion-0 {
    overflow-y: auto;
    height: 100%;
  }
}

<div
  className="emotion-3"
  data-testid="iamTheDataTestId"
  role="dialog"
  tabIndex={-1}
>
  <Content
    autoFocus={false}
    components={Object {}}
    isChromeless={false}
    isHeadingMultiline={true}
    onClose={[Function]}
    shouldCloseOnEscapePress={true}
    shouldScroll={true}
    stackIndex={0}
  >
    <div
      className="emotion-2"
    >
      <ModalHeader
        isHeadingMultiline={true}
        onClose={[Function]}
        showKeyline={false}
      />
      <Styled(div)
        className="emotion-1"
      >
        <div
          className="emotion-0"
        >
          <MyContent>
            <div>
              Hello
            </div>
          </MyContent>
        </div>
      </Styled(div)>
      <ModalFooter
        onClose={[Function]}
        showKeyline={false}
      />
      <TouchProvider>
        <ScrollLock
          accountForScrollbars={true}
        >
          <Sheet
            styles="body {
      box-sizing: border-box !important;
      overflow: hidden !important;
      position: relative !important;
      
      padding-right: 1024px !important;
    }"
          />
        </ScrollLock>
      </TouchProvider>
    </div>
  </Content>
</div>
`;

exports[`Using react-test-library Modal should be found by data-testid Using container snapshot 1`] = `<div />`;
